src/i18n/utils/load-language-async.ts

```diff
@@ -7,20 +7,25 @@ function setI18nLanguage(lang) {
   i18n.locale = lang;
   axios.defaults.headers.common["Accept-Language"] = lang;
   document.querySelector("html")?.setAttribute("lang", lang);
-  return lang;
 }
 
-export function loadLanguageAsync(lang) {
-  if (i18n.locale === lang || loadedLanguages.includes(lang)) {
-    return Promise.resolve(lang);
+export async function loadLanguageAsync(lang) {
+  if (i18n.locale === lang) {
+    return;
   }
 
-  // If the language hasn't been loaded yet
-  return import(
-    /* webpackChunkName: "lang-[request]" */ `@/i18n/messages/${lang}.json`
-  ).then((messages) => {
-    i18n.setLocaleMessage(lang, messages.default);
+  if (!loadedLanguages.includes(lang)) {
     loadedLanguages.push(lang);
-    return setI18nLanguage(lang);
-  });
+
+    try {
+      const messages = await import(
+        /* webpackChunkName: "lang-[request]" */ `@/i18n/messages/${lang}.json`
+      );
+      i18n.setLocaleMessage(lang, messages.default);
```

Comment:
The `messages.default` confused me at first because I thought that messages was already the imported object, but it turns out it's the imported module. [Here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#importing_defaults) I learned that you can directly get the default objects of dyanimc imports with the following syntax:

```suggestion
      const { default: messages } = await import(
        /* webpackChunkName: "lang-[request]" */ `@/i18n/messages/${lang}.json`
      );
      i18n.setLocaleMessage(lang, messages);
```

Reply:
Yea it's virtually the same thing but if you prefer I can use destructuring indeed!

Reply:
nahh it's good :wink: